### DISCUSSION TITLE
Error while validating the service descriptor 'ServiceType: Elsa.Services.IWorkflowPublisher Lifetime: Scoped ImplementationType:

### DISCUSSION THREAD

**Discussion by vabs212** [elsa-workflows/elsa-core#1402] [Q&A]
Some services are not able to be constructed (Error while validating the service descriptor 'ServiceType: Elsa.Services.IWorkflowPublisher Lifetime: Scoped ImplementationType: Elsa.Services.WorkflowPublisher': Unable to resolve service for type 'Elsa.Persistence.IWorkflowDefinitionStore' while attempting to activate 'Elsa.Services.WorkflowPublisher'.) (Error while validating the service descriptor 'ServiceType: Elsa.Activities.Email.Activities.SendEmail Lifetime: Transient ImplementationType:

**Comment by vabs212** [comment]
Hi Can someone please help me in resolving this error .Actually i am trying to make an application on elsa framework with the given example in link https://medium.com/c-sharp-progarmming/whats-new-in-elsa-2-0-ea7410b65eea. 
But on running the application i ma getting the above error .please help me in resolving this.
![image](https://user-images.githubusercontent.com/66916559/129891549-8d418a89-84d5-4cda-b046-54744c6d016f.png)

This is my startup

**Reply by sfmskywalker** [reply]
It looks like you might be mixing Elsa 1 with Elsa 2 packages. Make sure to only use Elsa packages that start with 2.2. Use the [quick starts](https://elsa-workflows.github.io/elsa-core/docs/next/quickstarts/quickstarts-console-hello-world) & [guides](https://elsa-workflows.github.io/elsa-core/docs/next/guides/guides-recurring-tasks) sections of the documentation website for examples written against Elsa 2.
